Refactor LDLib calls

omergunr100 opened the pull request:
## What
Refactored some rendering code and replaced the ldlib mod load check with the forge method.
Unldlibning wherever possible without losing my sanity.

## Implementation Details
Refactored some conditions in rendering code to use the appropriate methods.
Refactored mod load checks to use native forge methods instead of ldlib method.

## Outcome
Reduced LDLib Dependency

omergunr100 committed:
i didn't remember the axis methods when i wrote it ok
it was after a couple hours of fighting with the vertex methods

omergunr100 committed:
get offset timer should now work on both client and server again

omergunr100 committed:
added the getTime thing back for client

omergunr100 committed:
moved mod load checks to forge native method

omergunr100 committed:
doing the actual thing with the stuff

omergunr100 committed:
CLIENT_TIME

omergunr100 committed:
moved unsubscribe to onInvalid and removed check from subscriber runnable

omergunr100 committed:
getOffsetTimer in pbf

omergunr100 committed:
Here is my wisdom:
If your ball is too big for your mouth it's not yours.

omergunr100 committed:
feck

tomprince commented on src/main/java/com/gregtechceu/gtceu/api/machine/IMachineBlockEntity.java:
```diff
@@ -53,7 +54,14 @@ default void scheduleRenderUpdate() {
+            return getOffset();
+        } else if (level().isClientSide()) {
+            return GTValues.CLIENT_TIME + getOffset();
+        } else if (level().getServer() == null) {
```
Does this want to be `!= null`?

omergunr100 replied:
yes, when I split the middle condition I forgot to invert this part

omergunr100 committed:
thanks tom

omergunr100 committed:
Merge branch '1.20.1' into 1.20.1-rendering-cringe

omergunr100 committed:
back into the gtceu class

omergunr100 committed:
more GTCEu less ldlib/platform

omergunr100 committed:
javadoc

screret commented on src/main/java/com/gregtechceu/gtceu/GTCEu.java:
```diff
@@ -53,45 +65,145 @@ public static ResourceLocation appendId(String id) {
+    @Deprecated(forRemoval = true, since = "1.0.21")
+    public static boolean isHighTier() {
+        return GTCEuAPI.isHighTier();
+    }
```
Remove.

omergunr100 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1-rendering-cringe

# Conflicts:
#	src/main/java/com/gregtechceu/gtceu/integration/emi/GTEMIPlugin.java
#	src/main/java/com/gregtechceu/gtceu/integration/emi/recipe/GTRecipeEMICategory.java
#	src/main/java/com/gregtechceu/gtceu/integration/jei/GTJEIPlugin.java
#	src/main/java/com/gregtechceu/gtceu/integration/jei/recipe/GTRecipeJEICategory.java
#	src/main/java/com/gregtechceu/gtceu/integration/rei/GTREIPlugin.java
#	src/main/java/com/gregtechceu/gtceu/integration/rei/recipe/GTRecipeREICategory.java

omergunr100 committed:
new one found

krossgg commented on src/main/java/com/gregtechceu/gtceu/integration/jei/GTJEIPlugin.java:
```diff
@@ -69,7 +67,7 @@ public void registerCategories(@NotNull IRecipeCategoryRegistration registry) {
             registry.addRecipeCategories(new GTBedrockOreInfoCategory(jeiHelpers));
         for (GTRecipeCategory category : GTRegistries.RECIPE_CATEGORIES) {
-            if (category.shouldRegisterDisplays()) {
+            if (GTCEu.isDev() || category.shouldRegisterDisplays()) {
```
shouldRegisterDisplays already checks for dev env - please remove otherwise this bypasses the furnace recipe blacklist

omergunr100 committed:
cursed

omergunr100 committed:
refactor is mod loaded into a subclass

omergunr100 committed:
removed all ldlib from main gtceu class

omergunr100 committed:
removed incorrect merge result

omergunr100 committed:
rename

omergunr100 committed:
was working in runClient but not in production

krossgg commented on src/main/java/com/gregtechceu/gtceu/GTCEu.java:
```diff
@@ -53,45 +67,150 @@ public static ResourceLocation appendId(String id) {
+    public static class Mods {
+
+        public static boolean isJeiLoaded() {
+            return !(isEmiLoaded() || isReiLoaded()) && isModLoaded(GTValues.MODID_JEI);
```
Can you change this to specifically *check* if EMI or REI are loaded, not if we *think* they are loaded. This causes an issue where, if you turn off the EMI overlay while both JEI & EMI are installed, `isJEILoaded` will return true even though JEI isn't actually usable.

krossgg commented on src/main/java/com/gregtechceu/gtceu/GTCEu.java:
```diff
@@ -53,45 +67,150 @@ public static ResourceLocation appendId(String id) {
-        return GTCEuAPI.isHighTier();
+    public static class Mods {
+
+        public static boolean isJeiLoaded() {
```
make JEI REI EMI all caps pls :)))

krossgg commented on src/main/java/com/gregtechceu/gtceu/core/mixins/WorldLoaderMixin.java:
```diff
@@ -0,0 +1,31 @@
+    @Inject(method = "load",
+            at = @At(value = "INVOKE",
+                     target = "Lnet/minecraft/server/ReloadableServerResources;loadResources(Lnet/minecraft/server/packs/resources/ResourceManager;Lnet/minecraft/core/RegistryAccess$Frozen;Lnet/minecraft/world/flag/FeatureFlagSet;Lnet/minecraft/commands/Commands$CommandSelection;ILjava/util/concurrent/Executor;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;"))
+    private static <D, R> void gtceu$injectLoad(WorldLoader.InitConfig initConfig,
```
I think I found a way to do this that isn't COMPLETE Susge
This RegistryAccess object that we're trying to get is the exact same RegistryAccess that gets sent when the `addReloadListenerEvent` event is fired, and we can freely grab that one and store it.
Preliminary testing shows that it works fine without nulling out or crashing, and ore veins/fluid veins are still present. Seems to work server/client too.
See `ReloadableServerResources#loadResources`
Maybe ask screret.

omergunr100 replied:
it does make sense, I just finished following the origin of the frozen registry, it sends the exact same registry access object that the mixin was saving and that's the only place that triggers this event.

krossgg commented on src/main/java/com/gregtechceu/gtceu/GTCEu.java:
```diff
@@ -6,31 +6,45 @@
     public static final String NAME = "GregTechCEu";
     public static final Logger LOGGER = LoggerFactory.getLogger(NAME);
 
+    private static final RegistryAccess BLANK = RegistryAccess.fromRegistryOfRegistries(BuiltInRegistries.REGISTRY);
+    private static RegistryAccess FROZEN = BLANK;
+
```
Maybe just stuff all of this into `GTRegistries` and make everything use `GTRegistries#builtinRegistry` ?

omergunr100 committed:
capitalized properly mod names.
replace xei loaded checks in jei loaded check with just the isModLoaded check.
moved frozen registry shenanigans to GTRegistries and an event listener to replace the mixin (seems to be working fine).

krossgg commented on src/main/java/com/gregtechceu/gtceu/forge/ForgeCommonEventListener.java:
```diff
@@ -605,4 +605,9 @@ public static void remapIds(MissingMappingsEvent event) {
     }
+
+    @SubscribeEvent
+    public static void onAddReloadableServerResources(AddReloadListenerEvent event) {
```
Line 283 :ICANT:

omergunr100 committed:
what can I say, I guess I should take my adhd medicine more

screret commented on src/main/java/com/gregtechceu/gtceu/GTCEu.java:
```diff
@@ -53,45 +61,132 @@ public static ResourceLocation appendId(String id) {
+     * @return if it's safe to access the current instance {@link net.minecraft.world.level.Level Level} on client or if
+     *         it's safe to access any level on server.
+     */
+    public static boolean canGetServerLevel() {
```
why is this named `canGetServerLevel` but does client checks too?
also, is this safe to run on a dedicated server (because of aforementioned clientside class usage)?

omergunr100 replied:
it should be, we used to use `Platform.isServerNotSafe()` which is just the negation of this method.
Although I should try to run it on a server and see if anything breaks.

omergunr100 replied:
Should maybe rename it to `isLevelSafe()` or something to make it clearer

omergunr100 replied:
Tried to run this version on a server, built multiple multiblocks which formed, server didn't crash, shut it down and restarted it, still didn't crash, looks like it's working fine.

screret commented on src/main/java/com/gregtechceu/gtceu/api/registry/GTRegistries.java:
```diff
@@ -66,6 +67,9 @@ public final class GTRegistries {
     public static final GTRegistry.RL<DimensionMarker> DIMENSION_MARKERS = new GTRegistry.RL<>(
             GTCEu.id("dimension_marker"));
 
+    private static final RegistryAccess BLANK = RegistryAccess.fromRegistryOfRegistries(BuiltInRegistries.REGISTRY);
+    private static RegistryAccess FROZEN = BLANK;
+
```
I'd move these down to be after the `register` method.

omergunr100 committed:
moved registry access closer to related functions

screret committed:
Merge branch '1.20.1' into fork/omergunr100/1.20.1-rendering-cringe

# Conflicts:
#	src/main/java/com/gregtechceu/gtceu/client/ClientProxy.java
#	src/main/java/com/gregtechceu/gtceu/data/recipe/CraftingComponent.java
#	src/main/java/com/gregtechceu/gtceu/integration/map/GroupingMapRenderer.java
#	src/main/java/com/gregtechceu/gtceu/integration/map/WaypointManager.java
#	src/main/java/com/gregtechceu/gtceu/integration/map/xaeros/XaerosMapPlugin.java
